src/binary/cursor.rs

```diff
@@ -49,8 +145,12 @@ impl Default for EncodedValue {
             annotations: Vec::new(),
```

Comment:
I am not sure what the balance is here: We could make annotations an `Option<...>` and avoid the new Vec here but that will add extra code around any operation no annotations to check optionality... Food for though mostly

Reply:
I went with `Vec` because [`Vec::new()` doesn't allocate](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.new) (so it should be about the same expense as making an `Option<Vec<_>>`) and (as you mention) it simplifies the code in places where it's used.